hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OMDBCheckpointServlet.java

```diff
@@ -106,6 +132,37 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) {
       return;
     }
 
+    // Check ACL for dbCheckpoint only when global Ozone ACL is enable
+    if (om.getAclsEnabled()) {
+      final String remoteUser = request.getRemoteUser();
+      final java.security.Principal userPrincipal = request.getUserPrincipal();
+      if (userPrincipal == null) {
+        // Fallback to checking login user if userPrincipal is null.
+        // Note: In prod, a secure cluster would deploy Kerberos so this case
```

Comment:
Kerberos may or may not be used for web endpoints in secure cluster. There are multiple `AuthenticationHandler` which can be configured. Are you sure that userPrincipal can never be null in real case? Based on the Javadoc of `HttpServletRequest` `null` means un-authenticated. Wouldn't be more safe to update the unit test instead of doing this check in the producation code?

Is there any unit test which request this branch?

Reply:
Hmm, when Kerberos is enabled, if `request.getUserPrincipal()` is null, I don't see a way to get the principal name (`request.getUserPrincipal().getName()`) to that I can check against. Therefore, I assumed when Kerberos is enabled, `request.getUserPrincipal()` should not be null. And when I test this chunk of code manually locally (unsecured), `request.getUserPrincipal()` is indeed null.

When Kerberos is not in use, the code just check with `request.getRemoteUser()`.

Reply:
I agree I should add some tests. I'm going to add some lines in the ozone secure robot test.

Reply:
> Is there any unit test which require this branch?

Yes. `TestOMDbCheckpointServlet#testDoGet`, which doesn't have Kerberos but have ACL enabled.

Reply:
Looks like we had some robot tests for dbCheckpoint in `web.robot`:
https://github.com/apache/ozone/blob/4ad03188ed4fbbe8d6dce1e8e0c8d91518904fc1/hadoop-ozone/dist/src/main/smoketest/spnego/web.robot#L59-L60

in SPNEGO testing.

Reply:
> I agree. But since you mentioned Kerberos might not be in-use on a secure cluster, we would still need some sort of user name verification, no? If that's the case I can adjust the comment.

I think, it's fine as is, the only thing what I would suggest is to remove the unit test related part from the production code unless it's fully required (can be convinced, but let me know which test requires is). Just throw an exception when user is null.

Instead of this:
```
     final java.security.Principal userPrincipal = request.getUserPrincipal();
      if (userPrincipal == null) {
        // Fallback to checking login user if getUserPrincipal returns null.
        // Note: In prod, a secure cluster with Kerberos should not hit this
        // branch. This is mostly here for UT and dev testing.
 
        final String remoteUser = request.getRemoteUser();
         if (!hasPermission(remoteUser)) {
          LOG.error("Permission denied: Current login user '{}' does not have"
              + " access to /dbCheckpoint.", remoteUser);
          response.setStatus(HttpServletResponse.SC_FORBIDDEN);
          return;
        }
```

I would use this:
```
    final java.security.Principal userPrincipal = request.getUserPrincipal();
      if (userPrincipal == null) {
          LOG.error("Permission denied: Current login user '{}' does not have"
              + " access to /dbCheckpoint.", remoteUser);
          response.setStatus(HttpServletResponse.SC_FORBIDDEN);
          return;
        }
 }
 ```

Reply:
done